views/default/input/submit.php

```diff
@@ -5,10 +5,16 @@
  * @package Elgg
  * @subpackage Core
  *
- * @uses $vars['class'] CSS class that replaces elgg-button-submit
+ * @uses $vars['class'] Additional CSS class
  */
 
 $vars['type'] = 'submit';
-$vars['class'] = elgg_extract('class', $vars, 'elgg-button-submit');
+
+if (!isset($vars['class'])) {
+	$vars['class'] = array('elgg-button-submit');
+} elseif (!is_array($vars['class'])) {
+	$vars['class'] = array($vars['class']);
+	$vars['class'][] = 'elgg-button-submit';
+}
```

Comment:
i think this requires a seperate ticket as classes are handled as array in multiple ways, so i will create a seperate ticket for that and leave this in this current state